props/saber_BC_buttons.h

```diff
@@ -490,12 +490,31 @@ class SaberBCButtons : public PROP_INHERIT_PREFIX PropBase {
         thrust_begin_millis_ = millis();
       }
 
+    // Mute timer check - play optional mute.wav first.
+    if (mute_timer_.timerCheck()) {
+      if (SetMute(true)) {
+        unmute_on_deactivation_ = true;
+        TurnOnHelper();
+      }
+    }
     // Scroll Presets timer check - avoid beep/wav overlap
     if (scroll_presets_timer_.timerCheck() && scroll_presets_) {
         SaberBase::DoEffect(EFFECT_NEWFONT, 0);
     }
   }  // Loop()
 
+
+  float GetSoundLengthNow(Effect* effect) {
+    RefPtr<BufferedWavPlayer> tmp = GetWavPlayerPlaying(effect);
+    if (tmp) {
+      tmp->UpdateSaberBaseSoundInfo();
```

Comment:
This is only needed when a sound is started automatically started, and we want to update the SaberBaseSoundInfo after the fact.  When you call PlayCommon(), the sound info will already be updated.

Reply:
ah. yes that's easier, thanks. changes made.